lib/centralisation/ui/top_bar.dart

```diff
@@ -0,0 +1,107 @@
+import 'package:flutter/material.dart';
+import 'package:heroicons/heroicons.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:myecl/centralisation/tools/constants.dart';
+import 'package:myecl/drawer/providers/swipe_provider.dart';
+import 'package:myecl/centralisation/providers/centralisation_page_provider.dart';
+import 'package:myecl/centralisation/providers/openLink.dart';
+import 'package:myecl/centralisation/ui/pages/main.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+
+
+
+
+class TopBar extends HookConsumerWidget {
+  final SwipeControllerNotifier controllerNotifier;
+  const TopBar({Key? key, required this.controllerNotifier}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final page = ref.watch(centralisationPageProvider);
+    final pageNotifier = ref.watch(centralisationPageProvider.notifier);
+    final favorites = ref.watch(favoritesProvider);
+
+    return Column(
+      children: [
+        const SizedBox(
+          height: 15,
+        ),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            SizedBox(
+              width: 70,
+              child: Builder(
+                builder: (BuildContext appBarContext) {
+                  return IconButton(
+                    onPressed: () {
+                      switch (page) {
```

Comment:
I think you have missed copying some code of other top_bar because clicking on this button does not do anything, but it should trigger the drawer animation